pilot/pkg/proxy/envoy/v1/watcher.go

```diff
@@ -55,34 +56,48 @@ type CertSource struct {
 }
 
 type watcher struct {
-	agent    proxy.Agent
-	role     model.Node
-	config   meshconfig.ProxyConfig
-	certs    []CertSource
-	pilotSAN []string
+	agent         proxy.Agent
+	role          model.Node
+	config        meshconfig.ProxyConfig
+	requiredCerts []CertSource // These certs must present for Envoy to start
```

Comment:
Danger ! Why ? Envoy can operate in degraded mode, it's almost never the right thing to shut down a service just because some feature doesn't work.

Reply:
I don't think Envoy can work in degraded mode without loading the cert files. As we observed, if we don't have the certs ready, Envoy will always fail to start. And the pilot agent will keep restarting Envoy in an infinite loop.

Reply:
Why are the certs missing ? I'm pretty sure ingress starts without the https secrets, so envoy has this capability.

Reply:
The certs could be temporarily missing because the CA is not available.

Reply:
there are cases where not starting or crashing when unable to get required components is better, it gives a chance to k8s to try the container somewhere else etc